python/mirascope/llm/providers/openai/completions/provider.py

```diff
@@ -1,456 +1,71 @@
-
-        Args:
-            model_id: Model identifier to use.
-            messages: Messages to send to the LLM.
-            tools: Optional tools that the model may invoke.
-            format: Optional response format specifier.
-            **params: Additional parameters to configure output (e.g. temperature). See `llm.Params`.
-
-        Returns:
-            An `llm.StreamResponse` object for iterating over the LLM-generated content.
-        """
-        input_messages, format, kwargs = _utils.encode_request(
-            model_id=model_id,
-            messages=messages,
-            tools=tools,
-            format=format,
-            params=params,
-        )
-
-        openai_stream = self.client.chat.completions.create(
-            **kwargs,
-            stream=True,
-        )
-
-        chunk_iterator = _utils.decode_stream(openai_stream)
-
-        return StreamResponse(
-            provider_id=self.active_provider_id,
-            model_id=model_id,
-            provider_model_name=model_name(model_id, "completions"),
-            params=params,
-            tools=tools,
-            input_messages=input_messages,
-            chunk_iterator=chunk_iterator,
-            format=format,
-        )
-
-    def _context_stream(
-        self,
-        *,
-        ctx: Context[DepsT],
-        model_id: OpenAIModelId,
-        messages: Sequence[Message],
-        tools: Sequence[Tool | ContextTool[DepsT]]
-        | ContextToolkit[DepsT]
-        | None = None,
-        format: type[FormattableT] | Format[FormattableT] | None = None,
-        **params: Unpack[Params],
-    ) -> ContextStreamResponse[DepsT] | ContextStreamResponse[DepsT, FormattableT]:
-        """Generate an `llm.ContextStreamResponse` by synchronously streaming from the OpenAI ChatCompletions API.
+        """Initialize the OpenAI Completions provider.
 
         Args:
-            ctx: Context object with dependencies for tools.
-            model_id: Model identifier to use.
-            messages: Messages to send to the LLM.
-            tools: Optional tools that the model may invoke.
-            format: Optional response format specifier.
-            **params: Additional parameters to configure output (e.g. temperature). See `llm.Params`.
-
-        Returns:
-            An `llm.ContextStreamResponse` object for iterating over the LLM-generated content.
+            api_key: OpenAI API key. If not provided, uses OPENAI_API_KEY env var.
+            base_url: Optional custom base URL for the API.
+            wrapped_by_openai_provider: If True, sets provider_id to "openai" instead of "openai:completions".
+            effective_provider_id: Override the provider_id used in Response/AssistantMessage.
+                If provided, takes precedence over wrapped_by_openai_provider.
+            model_name_resolver: Custom function to extract model name for API calls.
+                Receives model_id, returns model name to send to API.
```

Comment:
I think we can actually just always split on the first slash and return the second half